server/app/auth/StoredFileAcls.java

```diff
@@ -25,8 +24,20 @@ public StoredFileAcls() {
   }
 
   @JsonCreator
-  public StoredFileAcls(@JsonProperty("programReadAcls") HashSet<String> programReadAcls) {
-    this.programReadAcls = checkNotNull(programReadAcls);
+  public StoredFileAcls(
+      @Nullable @JsonProperty("programReadAcls") HashSet<String> programReadAcls) {
+    // If the file was created before the migration to using StoredFileAcls,
+    // programReadAcls will be null on initial load. In this case we initialize
+    // the internal state of the ACLs to an empty collection so the migration
+    // task can populate them.
+    // It is safest to leave this code in, even after the migration, in the
+    // event that somehow a StoredFile is created without serializing an
+    // instance StoredFileAcls.
```

Comment:
I don't think having the first part of this comment is particularly useful given the second part basically makes allowances for unknown reasons that would cause it.

I'd just do something more like "If there aren't read acls for some reason use an empty acl" and log the unexpected situation.